Prevent ambiguous arguments to `PhpRenderer::render`

gsteel opened the pull request:
`PhpRenderer::render` should be called with `(ViewModel)` or `(string, iterable)`.

When called with `(ViewModel, iterable)`, an exception will now be thrown instead of silently ignoring the `iterable` representing the view variables.

This will be useful signalling to users because in previous versions, the view model and the given variables would have been merged.

gsteel committed:
Prevent ambiguous arguments to `PhpRenderer::render`

`PhpRenderer::render` should be called with `(ViewModel)` or `(string, iterable)`.

When called with `(ViewModel, iterable)`, an exception will now be thrown instead of silently ignoring the `iterable` representing the view variables.

This will be useful signalling to users because in previous versions, the view model and the given variables would have been merged.

Signed-off-by: George Steel <george@net-glue.co.uk>

Ocramius reviewed (approved):
From my PoV, this is OK, but I would also explore restricting this at type level, somehow

Xerkus commented on src/Renderer/PhpRenderer.php:
```diff
@@ -50,6 +52,13 @@ public function render(
+        if ($templateNameOrModel instanceof ModelInterface && ! $variablesEmpty) {
+            throw RenderingFailedException::becauseOfAmbiguousArgumentsToPhpRenderer();
+        }
+
```
I think it can be stricter and require null.
Something passed at all is indicative of a problem. It just happened to be empty at a time.

gsteel committed:
Instead of checking for empty variables, assert variables to be null when a model is given

If you're passing anything as the second argument here, you're doing it wrong.

Signed-off-by: George Steel <george@net-glue.co.uk>

Xerkus commented on src/Exception/RenderingFailedException.php:
```diff
@@ -101,4 +101,13 @@ public static function becauseOfAnUnknownPlugin(
+    public static function becauseOfAmbiguousArgumentsToPhpRenderer(): self
+    {
+        return new self(
+            'Passing both view model view variables to render is ambiguous. '
```
Typo?

```suggestion
            'Passing both view model and variables to render is ambiguous. '
```

gsteel committed:
Fix exception message typo

Signed-off-by: George Steel <george@net-glue.co.uk>
